pkg/scheduler/scheduler_test.go

```diff
@@ -1723,3 +1730,167 @@ func TestRequeueAndUpdate(t *testing.T) {
+				Obj()).
+			Obj(),
+	}
+	queues := []kueue.Queue{
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+			Spec: kueue.QueueSpec{
+				ClusterQueue: "fooCq",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "bar"},
+			Spec: kueue.QueueSpec{
+				ClusterQueue: "barCq",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "baz"},
+			Spec: kueue.QueueSpec{
+				ClusterQueue: "bazCq",
+			},
+		},
+	}
+	tests := []struct {
+		name          string
+		workloads     []kueue.Workload
+		wantWorkloads sets.String
+	}{
+		{
+			name:          "scheduler with empty clusterQueues",
+			workloads:     []kueue.Workload{},
+			wantWorkloads: sets.String{},
+		},
+		{
+			name: "scheduler with active clusterQueues",
+			workloads: []kueue.Workload{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "a",
+						CreationTimestamp: metav1.NewTime(now),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "foo"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "b",
+						CreationTimestamp: metav1.NewTime(now),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "bar"},
+				},
+			},
+			wantWorkloads: sets.NewString("a", "b"),
+		},
+		{
+			name: "scheduler with clusterQueues containing multi workloads",
+			workloads: []kueue.Workload{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "a1",
+						CreationTimestamp: metav1.NewTime(now),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "foo"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "a2",
+						CreationTimestamp: metav1.NewTime(now.Add(time.Hour)),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "foo"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "b",
+						CreationTimestamp: metav1.NewTime(now),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "bar"},
+				},
+			},
+			wantWorkloads: sets.NewString("a1", "b"),
+		},
+		{
+			name: "scheduler with inactive clusterQueues",
+			workloads: []kueue.Workload{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "a",
+						CreationTimestamp: metav1.NewTime(now),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "foo"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "b",
+						CreationTimestamp: metav1.NewTime(now),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "bar"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "c",
+						CreationTimestamp: metav1.NewTime(now.Add(time.Hour)),
+					},
+					Spec: kueue.WorkloadSpec{QueueName: "baz"},
+				},
+			},
+			wantWorkloads: sets.NewString("a", "b"),
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			c := cache.New(client)
+			manager := queue.NewManager(fake.NewClientBuilder().WithScheme(scheme).Build(), c)
+			for _, cq := range clusterQueues {
+				if err := c.AddClusterQueue(ctx, &cq); err != nil {
```

Comment:
If we move the tests back to where it comes, we'll have to mock the `AddClusterQueue`, it's quite complexity and we will have to export `AddClusterQueue` method, but I think it's unnecessary for we will never add clusterQueue to cache in manager.

Reply:
why do you need to mock this? you only need to mock `ClusterQueueActive` function, right?

Reply:
sg.